### PR TITLE
fix(module:core): fix the style of CDK conflicts

### PR DESCRIPTION
**Pull request by HsuanXyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2874

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/tree-select/nz-tree-select.module.ts [hunk 1]
```diff
@@ -2,13 +2,14 @@ import { OverlayModule } from '@angular/cdk/overlay';
...
 import { FormsModule } from '@angular/forms';
+import { NzOverlayModule } from '../core/overlay/nz-overlay.module';
 import { NzEmptyModule } from '../empty/nz-empty.module';
 import { NzIconModule } from '../icon/nz-icon.module';
 import { NzTreeModule } from '../tree/nz-tree.module';
 import { NzTreeSelectComponent } from './nz-tree-select.component';
 
 @NgModule({
-  imports     : [ CommonModule, OverlayModule, FormsModule, NzTreeModule, NzIconModule, NzEmptyModule ],
+  imports     : [ CommonModule, OverlayModule, FormsModule, NzTreeModule, NzIconModule, NzEmptyModule, NzOverlayModule ],
```
Comment (HsuanXyz) [inline components/tree-select/nz-tree-select.module.ts]: ⌛️ refactor(module:tree-select): refactor tree-select component (#2898)

File: components/core/overlay/nz-connected-overlay.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { CdkConnectedOverlay } from '@angular/cdk/overlay';
+import { Directive } from '@angular/core';
+import { NzOverlayConfig } from './nz-overlay-config';
+
+@Directive({
+  selector: '[cdkConnectedOverlay][nz-connected-overlay], [cdkConnectedOverlay][nzConnectedOverlay]'
```
Comment (vthinkxie) [inline components/core/overlay/nz-connected-overlay.ts]: should be `[cdkConnectedOverlay][nzConnectedOverlay]`

File: components/core/overlay/nz-connected-overlay.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+})
+export class NzConnectedOverlayDirective {
+
+  constructor(private cdkConnectedOverlay: CdkConnectedOverlay) {
+    const config = new NzOverlayConfig();
+    this.cdkConnectedOverlay.backdropClass = config.backdropClass as string; // This can only be string
```
Comment (vthinkxie) [inline components/core/overlay/nz-connected-overlay.ts]: this.cdkConnectedOverlay.backdropClass = 'nz-overlay-transparent-backdrop'

File: components/core/overlay/nz-overlay-config.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { OverlayConfig } from '@angular/cdk/overlay';
```
Comment (vthinkxie) [inline components/core/overlay/nz-overlay-config.ts]: remove this file

**Pull request merged**